new-frontiers-14/frontier-station-14
# Reduce power flickering by increasing APC supply tolerance to 15 kW.

## About the PR



Changes the max supply and tolerance of all APCs to 15 kW.

## Why / Balance


Makes power networks less sensitive to fluctuations.  Things like the artifact analyzer and bays of thrusters, when turned on and off, cause power fluctuations at the moment.  This will resolve that (with instantaneous power draw increases up to 15 kW per APC).

With any sort of intermediate power generation backing it, larger supplyRampTolerance values will allow for better handling immediate changes to power draw, which seems more suitable for Frontier gameplay, especially if we want to add more togglable machines.

The supply tolerance of the APC must be large enough to handle the change in power draw, or flickering will happen.  Try this by having an APC with a tolerance of 5 kW and a backing substation with a tolerance of 30 kW.  An analysis pad will flicker the power when enabled.

Note: at maximum power draw, a standard APC will now last only ~3.3 seconds without a power tree behind it, down from 5, though it can, on its own, power 15 kW worth of stuff, up from 10.  If this is deemed unacceptable, increasing base APC capacity to 75 kJ may be worthwhile, and have negligible effects (+2.5 spesos) on ship value.

## How to test


1. Build a setup with a generator, an APC, an analysis console, and an artifact analyzer pad.
2. Make sure the APC is connected to the generator, the generator is set to 15kW at minimum.
3. Make sure the analyzer and console are connected.
4. Spawn an artifact, start scanning it.  There should be no power flicker.
5. Unanchor the console.  Set the generator to 10 kW.
6. Reanchor the console, make sure the artifact registers, restart scanning.  There should be no power flicker on the screen or analyzer, and the APC should be draining.

## Media


The minimal set up from the test bench.  The analyzer can toggle on and off without causing the power net to flicker.
![image](https://github.com/user-attachments/assets/b6b595d3-7ed3-48ea-a119-ffca357d78bc)
Minor experiments: a standard APC providing power for a large load (one standard console and another edited to have 78 kW of power draw).  In this situation (which is the norm for ship operation), the effect of maxPowerSupply on the APC itself is _negligible_.
![image](https://github.com/user-attachments/assets/36ad806a-8fe9-49c5-ad47-140c26fee94c)

## Requirements



## Breaking changes


**Changelog**


:cl:
- tweak: APCs can support larger changes to power draw, and should not toggle when analyzing an artifact.
